modules/platform/nuxeo-platform-document-routing/nuxeo-routing-rest-api/src/test/java/org/nuxeo/ecm/restapi/server/WorkflowEndpointTest.java

```diff
@@ -1396,6 +1404,45 @@ public void testRejectTaskInSubWorkflow() throws IOException {
 
     }
 
+    /*
+     * NXP-29171
+     */
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.restapi.server.routing.test:test-specific-task-request-unmarshalling.xml")
+    public void testSpecificTaskRequestWithFetchedGroup() throws IOException {
+        DocumentModel note = RestServerInit.getNote(0, session);
+
+        // create workflow as Administrator
+        String workflowInstanceId;
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "/workflow",
+                getCreateAndStartWorkflowBodyContent("confirm", singletonList(note.getId())))) {
+            assertEquals(Response.Status.CREATED.getStatusCode(), response.getStatus());
+
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            workflowInstanceId = node.get("id").textValue();
+        }
+
+        // get tasks for child WF
+        String taskId;
+        try (CloseableClientResponse response = getResponse(RequestType.GET,
+                "/id/" + note.getId() + "/@" + WorkflowAdapter.NAME + "/" + workflowInstanceId + "/task")) {
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            assertEquals(1, node.get("entries").size());
+            taskId = node.get("entries").get(0).get("id").textValue();
+        }
+
+        NuxeoGroup group = userManager.getGroup("administrators");
+
+        String groupJson = MarshallerHelper.objectToJson(group, RenderingContext.CtxBuilder.get());
+        String body = String.format("{\"entity-type\":\"task\", \"id\":\"%s\", \"variables\":{\"assignees\":[%s]}}",
+                taskId, groupJson);
+        // assign it with a fetched entity
+        try (CloseableClientResponse response = getResponse(RequestType.PUT, "/task/" + taskId + "/confirm", body)) {
+            assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
```

Comment:
Is there any check we can add to validate that the `assignees` is `administrators`?

Reply:
Of course, good idea 👍